src/components/ha-icon-button.ts

```diff
@@ -28,11 +34,11 @@ export class HaIconButton extends LitElement {
   };
 
   protected render(): TemplateResult {
-    // Note: `ariaLabel` required despite the `mwc-icon-button` docs saying `label` should be enough
     return html`
       <mwc-icon-button
-        .ariaLabel=${this.label}
-        .title=${this.hideTitle ? "" : this.label}
+        aria-label=${ifDefined(this.label)}
+        title=${ifDefined(this.hideTitle ? undefined : this.label)}
```

Comment:
I don't see that in their code... Or in my tests

Reply:
Hmm.  You're right.  They set it as empty because `icon` is initialized as empty.  At many other places in their code they have:

```
aria-label="${ifDefined(this.ariaLabel)}"
```

So I must have been looking at another component.  Still, we should not set them as empty strings.  Let the user agents handle them as unset unless there is content.

Reply:
I assume you mean:

```
.ariaLabel=${this.label || undefined}
```

Sure, but that only matters if `this.label` is initialized to `""`, and then it becomes impossible to remove the empty title attribute without more code.

FWIW, other things to consider:
1. It looks like the new Material 3 buttons will indeed not set an empty `aria-label` when it's not set
2. Setting as attributes is safer.  Google could have just as easily defined the type as `string | null`, and even Typescript had to make their own decision on the null/undefined type.  The ARIA spec for the IDL mixin is not well-defined in that respect as it attempts to be agnostic of programming language.  A property could break in the future, whereas an attribute cannot.

Is there some advantage to using properties over attributes?